--- mt-daapd-0.2.4.2/src/Makefile.in	2008-04-19 22:17:14.000000000 +0200
+++ ../Makefile_new.in	2011-02-16 21:51:52.053612723 +0100
@@ -61,7 +61,7 @@ am__mt_daapd_SOURCES_DIST = main.c daapd
 	strcasestr.h strsep.c redblack.c redblack.h dynamic-art.c \
 	dynamic-art.h query.c query.h mDNS.c mDNSClientAPI.h \
 	mDNSDebug.h mDNSPosix.c mDNSUNP.c rend-posix.c \
-	mDNSPlatformFunctions.h rend-unix.c rend-osx.c rend-howl.c \
+	mDNSPlatformFunctions.h rend-unix.c rend-osx.c rend-howl.c xbmc_binding.c xbmc_binding.h \
 	ogg.c rend-avahi.c
 @COND_REND_POSIX_TRUE@am__objects_1 = mDNS.$(OBJEXT) \
 @COND_REND_POSIX_TRUE@	mDNSPosix.$(OBJEXT) mDNSUNP.$(OBJEXT) \
